app/containers/AdminContainer.js

```diff
@@ -0,0 +1,117 @@
+import React from 'react'
+import fetcher from 'utils/fetcher'
+import Loading from 'components/Loading'
+import Error from 'components/Error'
+import Breadcrumbs from 'components/Breadcrumbs'
+import AllowEditButton from 'components/AllowEditButton'
+import Modal from 'components/Modal'
+import { downloadJSON } from 'utils/download'
+
+export default class AdminContainer extends React.Component {
+  state = {
+    loaded: false,
+    error: null,
+    docId: null,
+    adminConfig: {},
+    showCreateAdminModal: false,
+    adminName: '',
+    adminPassword: ''
+  }
+
+  static getDerivedStateFromProps (nextProps, prevState) {
+    const docId = decodeURIComponent(nextProps.match.path.split('/:couch/')[1])
+    return { ...prevState, docId }
+  }
+
+  async componentDidMount () {
+    const { couchUrl } = this.props
+    const { docId } = this.state
+    try {
+      const adminConfig = await fetcher.get(`${couchUrl}${docId}`)
+      this.setState({ adminConfig, loaded: true })
+    } catch (error) {
+      this.setState({ error, loaded: true })
+      console.error(error)
+    }
+  }
+
+  onSubmitNewAdmin = async event => {
+    event.preventDefault()
+    const {couch, couchUrl} = this.props
+    const {adminName, docId, adminConfig, adminPassword} = this.state
+    if (!adminName || !adminPassword) {
+      window.alert('admin name and password required')
+      return
+    }
+    window.alert('downloading current admin list as backup')
```

Comment:
any reason for backing up before updating/creating(not deleting), just curious

Reply:
not really actually for add, i was thinking more when we add changing more of the config as a little helper against whoopsies (e.g. you deleted the wrong admin & wanna check which it was)